src/group/group.repository.ts

```diff
@@ -48,11 +48,41 @@ export default class GroupRepository {
   }
 
   /**
-   * Finds groups by partial name using regex.
-   * @param partialName - the partial name of the group.
+ * Finds public groups by partial name / tag using regex.
+ * @param partial - a string that can be a partial name/tag of the group.
+ */
+  static searchPublicByNameAndTag(partial: string): Promise<IGroup[]> {
+    return groupModel.find({
+      $and: [
+        { type: GroupType.Public },
+        {
+          $or: [
+            { name: { $regex: partial, $options: 'i' } },
+            { 'tags.label': { $regex: partial, $options: 'i' } },
+          ],
+        },
+      ],
+    }).exec();
+  }
+
+  /**
+   * Finds private groups by user ID and partial name / tag using regex.
+   * @param userID - an ID of a user that mut be in the returned groups.
+   * @param partial - a string that can be a partial name/tag of the group.
    */
-  static searchByName(partialName: string): Promise<IGroup[]> {
-    return groupModel.find({ name: { $regex: partialName, $options: 'i' } }).exec();
+  static searchPrivate(userID: string, partial: string): Promise<IGroup[]> {
```

Comment:
Change the function's name to searchPrivateByNameAndTag

Reply:
It also search by `userID`, but the name `searchPrivateByNameTagAndUser` seems to bulky.
I think I should better change the name of the public search to just `searchPublic`.

Reply:
I think that searchPrivateByNameAndTag is enough because we have to include userId when searching for private groups. But changing to searchPublic is a good option too.